nntrainer/dataset/databuffer.cpp

```diff
@@ -251,11 +250,16 @@ DataBuffer::startFetchWorker(const std::vector<TensorDim> &input_dims,
 
   return std::async(std::launch::async, [bq, generator] {
     while (true) {
-      auto iteration = generator();
-      auto last = std::get<0>(iteration);
-      bq->wait_and_push(std::move(iteration));
-      if (last == true) {
-        break;
+      try {
+        auto iteration = generator();
+        auto last = std::get<0>(iteration);
+        bq->wait_and_push(std::move(iteration));
```

Comment:
The validation checks for the data and its shape will be performed here? as the iteration is coming from external functions. If yes, please add TODO here or at appropriate location.
These checks will probably be removed when the data is pre-allocated.

Reply:
I am currently working on the pre-allocate scenario, so let me add a comment here instead.